src/devices/imagedev/floppy.cpp

```diff
@@ -1707,9 +1707,10 @@ void floppy_sound_device::sound_stream_update(sound_stream &stream, std::vector<
 		{
 			idx = m_spin_playback_sample;
 			sampleend = m_sample[idx].data.size();
-			out = m_sample[idx].data[m_spin_samplepos++];
-
-			if (m_spin_samplepos >= sampleend)
+			m_spin_samplepos++;
+			if (m_spin_samplepos < sampleend)
+				out = m_sample[idx].data[m_spin_samplepos];
+			else
```

Comment:
This changes behavior by effectively replacing a postincrement with a preincrement.

Reply:
This was tested using -wavwrite and feeding the before/after files into Audacity. The result was identical as far as I could tell by inverting one track, then mix and render, producing silence and a flat waveform. I tried just now to leave the bounds check in there and do the post increment the way it was and I get very slightly different before/after audio through Audacity.

Reply:
The output *should* be different because there won't be invalid samples in the output anymore.

Just do this:
if (m_spin_samplepos < sampleend)
    out = m_sample[idx].data[m_spin_samplepos++];
else